R/quosure-utils.R

```diff
@@ -0,0 +1,28 @@
+#' Copy environment content to the other environment
+#' To avoid sharing the same environment by multiple `Quosure` objects
+#' @keywords internal
+.copy_env <- function(env) {
+  # need to copy the objects from old env to new env
+  # to avoid quosure objects sharing the same environment
+  env_new <- new.env(parent = parent.env(.GlobalEnv))
+  for (i in ls(env)) {
+    env_new[[i]] <- env[[i]]
+  }
+  env_new
+}
+
+#' Combine two code vectors and keep unique names
+#'
+#' Combines two code vectors and modifies duplicated names.
+#' @param x (`character`) vector of code.
+#' @param y (`character`) vector of code.
+#' @return `character` containing both `x` and `y` combined with
+#'  unique names.
+#' @keywords internal
+.keep_code_name_unique <- function(x, y = character(0)) {
```

Comment:
keep_name_unique - this name indicates that the main responsibility is to _mutate names_ whereas the main responsibility here is to make _combine_ action.
I suggest to either rename it into c_with_unique_names or split it into two steps: (i) c, (ii) keep unique names.

Reply:
marking as resolved 
- PR here: https://github.com/insightsengineering/teal.code/pull/68